### PR TITLE
Upgrade to Amazon Linux 2

### PR DESCRIPTION
**Pull request by mattburgess**

### CODE REVIEW

File: bootstrap_actions/cloudwatch.sh [hunk 1]
```diff
@@ -25,107 +25,111 @@ cat > /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json <<CWAGEN
...
     "force_flush_interval" : 15
   }
 }
 CWAGENTCONFIG
 
+%{ if emr_release == "5.29.0" ~}
 # Download and install CloudWatch Agent
-curl https://s3.${AWS_DEFAULT_REGION}.amazonaws.com/amazoncloudwatch-agent-${AWS_DEFAULT_REGION}/centos/amd64/latest/amazon-cloudwatch-agent.rpm -O
+curl https://s3.$${AWS_DEFAULT_REGION}.amazonaws.com/amazoncloudwatch-agent-$${AWS_DEFAULT_REGION}/centos/amd64/latest/amazon-cloudwatch-agent.rpm -O
 rpm -U ./amazon-cloudwatch-agent.rpm
```
Comment (ChrisScottThomas) [inline bootstrap_actions/cloudwatch.sh:128]: Is this needed?

File: bootstrap_actions/cloudwatch.sh [hunk 2]
```diff
@@ -25,107 +25,111 @@ cat > /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json <<CWAGEN
...
 # To maintain CIS compliance
 usermod -s /sbin/nologin cwagent
 
 start amazon-cloudwatch-agent
+%{ else ~}
+systemctl start amazon-cloudwatch-agent
+%{ endif ~}
```
Comment (ChrisScottThomas) [inline bootstrap_actions/cloudwatch.sh:135]: This doesn't work.
Should be something like...
`sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -c ssm:${cloudwatch_agent_config_ssm_parameter} -s`

File: bootstrap_actions/cloudwatch.sh (hunk 1 above)
Comment (mattburgess) [reply bootstrap_actions/cloudwatch.sh:128]: It's what was in there before. So...yes.

Comment (ChrisScottThomas) [reply bootstrap_actions/cloudwatch.sh:128]: But it's baked into the image, so you don't need to install it anymore.

**Review by ChrisScottThomas — APPROVED** [review]

**Pull request merged**